design/velero-upgrade.md

```diff
@@ -0,0 +1,80 @@
+# Velero Upgrade Design  
+
+## Background
+
+Besides a fresh installation, users also require to upgrade from old version with preserving the existing tasks (i.e., backup schedules) and configurations (i.e., Velero server & daemonset parameters). At present, the recommended Velero upgrade approach is finished manually with below steps:  
+- Update all CRDs by running ```velero install --crds-only --dry-run -o yaml | kubectl apply -f -```
+- Update Velero server and daemonset image by running ```kubectl set image```  
+
+For a complete example of the existing upgrade approach, refer to the Velero document at https://velero.io/docs/v1.9/upgrade-to-1.9/.
+
+The manual approach works for the existing scenarios, however, if we want to do some complex tasks during upgrade (i.e., installing a new module or changing/adding complex parameters), it will be very unfriendly in the user experiences.  
+For an concrete example, in v1.10, we have done some refactors to Velero's pod volume backup (a.k.a Restic backup formerly), two of the refactors are:  
+- Rename some Velero server and daemonset parameters
+- Rename Velero daemonset from "restic" to "node-agent"
+
+As a result, the existing upgrade approach is not sufficient:
+- If we still use manual steps, we have to tell users the detailed parameters that have been renamed, and users then needs to check their existence and modify them one by one.
+- If we still use manual steps, users have to delete the old restic daemonset and run ```velero install``` with the full parameters they specified during the fresh installation, because for one thing, Kubernetes doesn't support to rename a resource; for another, Velero doesn't provide a way to install a single resource only. 
+
+## Solution
+
+We want to add a new command besides ```velero install``` called ```velero upgrade```. In a upgrade case, users could call ```velero upgrade``` without bothering to know the details. This bring below benefits:  
+- Solve the upgrade problems in v1.10
+- Make Velero more user friendly, because the upgrade process is significantly simplified
+- Support future evolvements. For example, in future, Velero will have various data movement features, for which we will add new moudles inevitably. Then we can add enhancement incrementally to ```velero upgrade```, because inside the command, we can do whatever complex things 
+
+The ```velero upgrade``` command will fully replace the existing manual steps in the aforementioned Velero document.
```

Comment:
I assume the upgrade does not support any parameters like 
`velero upgrade --image`

It only supports hard coded?  If the answer is yes, how do we test the upgrade before the GA image is pushed can be a little tricky I hope we can add more detail in this design to clarify that.

Reply:
Added explanations for under-release Velero client case.